extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/HybridJacksonPool.java

```diff
@@ -0,0 +1,141 @@
+package io.quarkus.jackson.runtime;
+
+import com.fasterxml.jackson.core.util.BufferRecycler;
+import com.fasterxml.jackson.core.util.BufferRecyclerPool;
+import org.jctools.queues.MpmcUnboundedXaddArrayQueue;
+import org.jctools.util.Pow2;
+import org.jctools.util.UnsafeAccess;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Predicate;
+
+public class HybridJacksonPool implements BufferRecyclerPool {
+
+    static final BufferRecyclerPool INSTANCE = new HybridJacksonPool();
+
+    private HybridJacksonPool() { }
+
+    private static final Predicate<Thread> isVirtual = findIsVirtual();
+
+    private static Predicate<Thread> findIsVirtual() {
+        try {
+            MethodHandle virtualMh = MethodHandles.publicLookup().findVirtual(Thread.class, "isVirtual", MethodType.methodType(boolean.class));
```

Comment:
I would save the MethodHandle in a static final variable or sadly it would be very costy